flood_adapt/dbs_classes/interface/database.py

```diff
@@ -0,0 +1,147 @@
+import os
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Optional, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+from cht_cyclones.tropical_cyclone import TropicalCyclone
+
+from flood_adapt.dbs_classes.interface.element import AbstractDatabaseElement
+from flood_adapt.dbs_classes.interface.static import IDbsStatic
+from flood_adapt.object_model.hazard.interface.events import IEvent
+from flood_adapt.object_model.interface.benefits import IBenefit
+from flood_adapt.object_model.interface.site import Site
+
+
+class IDatabase(ABC):
+    base_path: Path
+    input_path: Path
+    output_path: Path
+    static_path: Path
```

Comment:
is this really useful if the database is a singelton? if that is the case there can only be one by definition so I'm not sure what having this interface adds.

Reply:
1. I still dream of a world where we dont have it as a singleton and have an architecture where its not needed.

2. I want to be able to have type checkers that actually provide useful info. due to cyclic imports and problematic project structure, its not always possible to import the full Database class.

Reply:
That's fair, I think we can leave it like now, and work on the database like we talked some other time